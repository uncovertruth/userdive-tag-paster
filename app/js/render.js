/* @flow */
'use strict'
import Vue from 'vue'
import Info from '../components/info'

export default (userData: object, onClick: Function) => {
  const vm = new Vue({
    el: '#info',
    methods: {},
    render: h => {
      return h(
        Info,
        {
          props: {
            pageInfo: userData
          }
        }
      )
    }
  })
  vm.$on('changeStatus', () => onClick())
}
