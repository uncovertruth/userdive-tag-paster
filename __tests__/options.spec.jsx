/* @flow */
/** @jsx h */
import { h } from 'preact'
import render from 'preact-render-to-string'

import Options from '../app/components/options'

describe('options', () => {
  const data = {
    id: 'id',
    env: 'env',
    host: 'host',
    ignore: 'ignore'
  }

  test('vaild configs', () => {
    const tree = render(<Options {...data} />)
    expect(tree).toMatchSnapshot()
  })
})
