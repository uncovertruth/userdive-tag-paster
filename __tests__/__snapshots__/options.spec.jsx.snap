// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`options vaild configs 1`] = `"<div><div><span>state: OFF</span></div><div><span>Id</span><input type=\\"text\\" value=\\"id\\" /></div><div><span>Env</span><input type=\\"text\\" value=\\"env\\" /></div><div><span>Source</span><input type=\\"text\\" value=\\"host\\" /></div><div><span>Ignored domains (plz split<b>line break)</b><br /></span><textarea rows=\\"4\\" cols=\\"40\\" value=\\"ignore\\"></textarea></div><button id=\\"toggle\\">Save</button></div>"`;
